src/main/java/uk/gov/register/presentation/resource/Pagination.java

```diff
@@ -4,13 +4,15 @@
 import javax.ws.rs.BadRequestException;
 import java.util.Optional;
 
-class Pagination {
+public class Pagination {
 
     private final long pageIndex;
+    private final String resourcePath;
     private final long totalEntries;
     private final long pageSize;
 
-    Pagination(Optional<Long> pageIndex, Optional<Long> pageSize, long totalEntries) {
+    Pagination(String resourcePath, Optional<Long> pageIndex, Optional<Long> pageSize, long totalEntries) {
```

Comment:
For pagination class it is number of totalEntries only. you may chose to pass estimated entries or total entries from the caller. total entries in pagination does not mean the number of records in database, it means total entries to paginate on.